xbmc/ApplicationPlayer.h

```diff
@@ -97,85 +86,101 @@ class CApplicationPlayer
-  bool  IsPausedPlayback();
-  bool  IsPassthrough() const;
-  bool  IsPlaying() const;
-  bool  IsPlayingAudio() const;
-  bool  IsPlayingVideo() const;
-  bool  IsPlayingGame() const;
-  bool  IsPlayingRDS() const;
-  void  LoadPage(int p, int sp, unsigned char* buffer);
-  bool  OnAction(const CAction &action);
-  void  OnNothingToQueueNotify();
-  void  Pause();
-  bool  QueueNextFile(const CFileItem &file);
-  void  Seek(bool bPlus = true, bool bLargeStep = false, bool bChapterOverride = false);
-  int   SeekChapter(int iChapter);
-  void  SeekPercentage(float fPercent = 0);
-  bool  SeekScene(bool bPlus = true);
-  void  SeekTime(int64_t iTime = 0);
-  void  SeekTimeRelative(int64_t iTime = 0);
-  void  SetAudioStream(int iStream);
-  void  SetAVDelay(float fValue = 0.0f);
-  void  SetDynamicRangeCompression(long drc);
-  void  SetMute(bool bOnOff);
-  bool  SetPlayerState(const std::string& state);
-  void  SetSubtitle(int iStream);
-  void  SetSubTitleDelay(float fValue = 0.0f);
-  void  SetSubtitleVisible(bool bVisible);
-  void  SetTime(int64_t time);
-  void  SetTotalTime(int64_t time);
-  void  SetVideoStream(int iStream);
-  void  SetVolume(float volume);
-  void  SetSpeed(float speed);
+  bool HasAudio() const;
+  bool HasMenu() const;
+  bool HasVideo() const;
+  bool HasGame() const;
+  bool HasRDS() const;
+  bool IsCaching() const;
+  bool IsInMenu() const;
+  bool IsPaused();
+  bool IsPausedPlayback();
+  bool IsPassthrough() const;
+  bool IsPlaying() const;
+  bool IsPlayingAudio() const;
+  bool IsPlayingVideo() const;
+  bool IsPlayingGame() const;
+  bool IsPlayingRDS() const;
+  void LoadPage(int p, int sp, unsigned char* buffer);
+  bool OnAction(const CAction &action);
+  void OnNothingToQueueNotify();
+  void Pause();
+  bool QueueNextFile(const CFileItem &file);
+  void Seek(bool bPlus = true, bool bLargeStep = false, bool bChapterOverride = false);
+  int SeekChapter(int iChapter);
+  void SeekPercentage(float fPercent = 0);
+  bool SeekScene(bool bPlus = true);
+  void SeekTime(int64_t iTime = 0);
+  void SeekTimeRelative(int64_t iTime = 0);
+  void SetAudioStream(int iStream);
+  void SetAVDelay(float fValue = 0.0f);
+  void SetDynamicRangeCompression(long drc);
+  void SetMute(bool bOnOff);
+  bool SetPlayerState(const std::string& state);
+  void SetSubtitle(int iStream);
+  void SetSubTitleDelay(float fValue = 0.0f);
+  void SetSubtitleVisible(bool bVisible);
+  void SetTime(int64_t time);
+  void SetTotalTime(int64_t time);
+  void SetVideoStream(int iStream);
+  void SetVolume(float volume);
+  void SetSpeed(float speed);
   bool SupportsTempo();
 
   CVideoSettings GetVideoSettings();
   void SetVideoSettings(CVideoSettings& settings);
 
-  protected:
-    std::shared_ptr<IPlayer> GetInternal() const;
+  CSeekHandler& GetSeekHandler();
+protected:
+  std::shared_ptr<IPlayer> GetInternal() const;
+
+private:
+  std::shared_ptr<IPlayer> m_pPlayer;
+  std::unique_ptr<CSeekHandler> m_seekHandler;
```

Comment:
why is this a pointer and not a plain memeber?

Reply:
No reason, will change it